src/mailmap.rs

```diff
@@ -0,0 +1,215 @@
+  /// @throws An error if the operation failed.
+  ///
+  /// @category Mailmap/Methods
+  ///
+  /// @signature
+  /// ```ts
+  /// class Mailmap {
+  ///   addEntry(entry: AddMailmapEntryData): void;
+  /// }
+  /// ```
+  #[napi]
+  pub fn add_entry(&mut self, entry: AddMailmapEntryData) -> crate::Result<()> {
+    self.inner.add_entry(
+      entry.real_name.as_deref(),
+      entry.real_email.as_deref(),
+      entry.replace_name.as_deref(),
+      &entry.replace_email,
+    )?;
+    Ok(())
+  }
+
+  /// Resolve a signature to its canonical form using a mailmap.
+  ///
+  /// Returns a new signature with the canonical name and email.
+  ///
+  /// @param {SignaturePayload} signature - Signature to resolve
+  /// @returns The resolved signature with canonical name and email
+  ///
+  /// @category Mailmap/Methods
+  ///
+  /// @signature
+  /// ```ts
+  /// class Mailmap {
+  ///   resolveSignature(signature: SignaturePayload): Signature;
+  /// }
+  /// ```
+  #[napi]
+  pub fn resolve_signature(&self, signature: SignaturePayload) -> crate::Result<Signature> {
+    let git_signature = git2::Signature::try_from(Signature::try_from(signature)?)?;
+
+    let resolved = match &self.inner {
+      MailmapInner::Repo(repo) => repo.resolve_signature(&git_signature)?,
+      MailmapInner::Owned(owned) => owned.resolve_signature(&git_signature)?,
+    };
+
+    Signature::try_from(resolved)
+  }
+}
+
+#[napi]
+impl Commit {
+  #[napi]
+  /// Get the author of this commit, using the mailmap to map it to the canonical name and email.
+  ///
+  /// @category Commit/Methods
+  ///
+  /// @signature
+  /// ```ts
+  /// class Commit {
+  ///   authorWithMailmap(mailmap: Mailmap): Signature;
+  /// }
+  /// ```
+  ///
+  /// @param {Mailmap} mailmap - The mailmap to use for mapping
+  /// @returns Author signature of this commit with mapping applied
+  pub fn author_with_mailmap(&self, mailmap: &Mailmap) -> crate::Result<Signature> {
+    let git_signature = match &mailmap.inner {
+      MailmapInner::Repo(repo) => self.inner.author_with_mailmap(repo)?,
+      MailmapInner::Owned(owned) => self.inner.author_with_mailmap(owned)?,
+    };
+    let signature = Signature::try_from(git_signature)?;
+    Ok(signature)
+  }
+
+  #[napi]
+  /// Get the committer of this commit, using the mailmap to map it to the canonical name and email.
+  ///
+  /// @category Commit/Methods
+  ///
+  /// @signature
+  /// ```ts
+  /// class Commit {
+  ///   committerWithMailmap(mailmap: Mailmap): Signature;
+  /// }
+  /// ```
+  ///
+  /// @param {Mailmap} mailmap - The mailmap to use for mapping
+  /// @returns Committer signature of this commit with mapping applied
+  pub fn committer_with_mailmap(&self, mailmap: &Mailmap) -> crate::Result<Signature> {
+    let git_signature = match &mailmap.inner {
+      MailmapInner::Repo(repo) => self.inner.committer_with_mailmap(repo)?,
+      MailmapInner::Owned(owned) => self.inner.committer_with_mailmap(owned)?,
+    };
```

Comment:
```suggestion
    let git_signature = self.inner.committer_with_mailmap(&mailmap.inner)?;
```
I think this can be simplified.